tests/jeos/check_efi_setup.pm

```diff
@@ -0,0 +1,228 @@
+    my $backend        = get_required_var('BACKEND');
+    my $blk_dev_driver = {
+        qemu  => 'virtblk',
+        svirt => {
+            xen    => 'xvd',
+            hyperv => 'scsi'
+        }
+    };
+
+    return ($backend eq 'qemu') ? $blk_dev_driver->{$backend} :
+      $blk_dev_driver->{$backend}->{get_required_var('VIRSH_VMM_FAMILY')};
+}
+
+# return a slice of parted's output
+sub extract_from_parted {
+    my $p_rec = shift;
+
+    $p_rec =~ s/,\s?/_/g;
+    local $/ = ';';
+    chomp $p_rec;
+    return (split /:/, $p_rec)[@_];
+}
+
+sub get_esp_from_parted {
+    my @parted_data = split /\n/, script_output 'parted --list --machine --script';
+    # locate drive record_info
+    # old dos partition table is not supported in JeOS
+    my $drive_raw_record = (grep(/gpt/ && qr/virtualized_block_type()/, @parted_data))[0];
+    ($drive_raw_record) or die "No gpt table found!\n";
+
+    # find possible boot partitions in JeOS image
+    my $boot_esp;
+    map {
+        my $p_no = extract_from_parted($_, 0);
+        $boot_esp->{part_no}   = $p_no;
+        $boot_esp->{drive}     = extract_from_parted($drive_raw_record, 0);
+        $boot_esp->{partition} = $boot_esp->{drive} . $boot_esp->{part_no};
+    } grep(/boot,\s?esp/, @parted_data);
+
+    assert_script_run "parted --script $boot_esp->{drive} align-check optimal $boot_esp->{part_no}";
+
+    return $boot_esp;
+}
+
+sub run {
+    my $self = shift;
+    $self->select_serial_terminal;
+
+    my $esp_details = get_esp_from_parted;
+    my $pkgs        = 'efivar mokutil';
+    $pkgs .= ' dosfstools' if (is_leap '<15.2');
+    zypper_call "in $pkgs";
+
+    die "Image did not boot in UEFI mode!\n"
+      unless ($esp_details || is_efi_boot || get_required_var('UEFI'));
+
+    # store ESP's file system and mount point
+    my ($efi_p_fs, $efi_p_mp) = split(/\s+/,
+        script_output 'df --output=fstype,target,source --local | grep ' . $esp_details->{partition});
+    ($efi_p_fs && $efi_p_mp && $esp_details->{partition}) or die "No mounted ESP partition was not found!\n";
+
+    record_info "esp [$esp_details->{partition}]", "\nfs [$efi_p_fs],\nmp [$efi_p_mp]";
+
+    # run fs check on ESP
+    assert_script_run "umount $efi_p_mp";
+    assert_script_run "fsck.vfat -tvV $esp_details->{partition}";
+    assert_script_run "mount $efi_p_mp";
+
+    # enable verbosity in shim
+    assert_script_run 'mokutil --set-verbosity true';
+
+    ## default boot
+    check_efi_state $efi_p_mp;
+    check_mok;
+    ## set gfx payload
+    set_grub_gfxmode;
+    assert_script_run "grub2-script-check --verbose ${ \GRUB_CFG }";
+    $self->rebooting('gfxmode set');
+    check_efi_state $efi_p_mp;
+    check_mok;
+    ## shim install
+    assert_script_run 'shim-install';
+    $self->rebooting('After shim-install');
+    check_efi_state $efi_p_mp;
+    check_mok;
+    # grub2 install
+    #assert_script_run "grub2-install --efi-directory=$efi_p_mp --target=x86_64-efi $esp_details->{drive}";
```

Comment:
I know that PR has WIP label so it is ok . But I would leave a comment as a reminder not to forget to drop this before commit :)

Reply:
Yes, we cannot forget about this. I have to clarify this part. According to [yast](https://github.com/yast/yast-bootloader/blob/master/src/lib/bootloader/grub_install.rb#L81), this is the way how to setup efi without SB. I experimented with it a bit, but it failed everywhere. But maybe I have just missed some parameter, if so we can fix. Otherwise I plan to drop it. Or we can replace it by modification of `/etc/sysconfig/bootloader` where we can set `SECURE_BOOT="no"` and use pbl instead of `grub2-install`. Personally I would prefer use generic grub2 tools instead of `pbl`.

Reply:
Are you sure you tested this with secure boot disabled? The command looks good to me.

Reply:
> Are you sure you tested this with secure boot disabled? The command looks good to me.

Yes, I haven't modified `/etc/sysconfig/bootloader`. I thought that `grub2-mkconfig` would not touch it. I will update the code. Thanks!

Reply:
grub2-mkconfig neither reads nor modifies it. I'm asking about the VM settings. If secure boot is enabled (no matter which keys) `grub2-install` will not result in a bootable system.

Reply:
![DeepinScreenshot_select-area_20200420094317](https://user-images.githubusercontent.com/35097794/79726687-79c37700-82eb-11ea-82e2-1e74f5774a0b.png)
Got mislead by the label. We would need to change the VM secure boot settings from the test depending on the backend.